app/controllers/registrations_controller.rb

```diff
@@ -19,8 +20,16 @@ def create
 
   private
 
+  def redirect_if_unsupported
+    zone = Zone.find_by(zipcode: zipcode)
+
+    if zone.nil?
+      redirect_to new_zone_subscription_url(zipcode)
+    end
+  end
+
   def zone
-    Zone.find_by!(zipcode: params[:zone_id])
+    Zone.find_by!(zipcode: zipcode)
   end
```

Comment:
This code will end up looking for zone twice if the action is successful (a zone is present). Would it work to memoize? The redirect saves you from needing the bang method later unless you're worried zones will be deleted from the DB.